### PR TITLE
Drawer cookbook recipe with navigation

### LINKED ISSUE #1518: Add Navigation to 'Add a Drawer' to a screen
From URL: https://flutter.io/cookbook/design/drawer/

Are there any plans to extend this cookbook example to include navigation? Navigation from a drawer is an extremely popular design pattern and it is difficult to find any good documentation on how to implement this design pattern. I wouldn't mind doing this myself but I am one of the people looking for guidance on how to implement this pattern.

### PR DESCRIPTION
**Pull request by miquelbeltran**
_Description of what this PR is changing or adding, and why:_

In this PR I extend the example found in the cookbook recipe for the Drawer, as proposed in the ticket issue #1518.

I used the documentation for the BottomNavigationBar as starting point and modified it a bit to fit a Drawer. From https://api.flutter.dev/flutter/material/BottomNavigationBar-class.html

I've also added a short explanation and an alert note that recommends the reader to check the navigation cookbook recipes for more information on how to implement navigation.

I hosted a working example here: https://mb-flutter-dev-1.web.app/cookbook/design/drawer

Video recording:

https://github.com/flutter/website/assets/2494376/48e4cc68-0b2d-4551-8414-2923684bc317

_Issues fixed by this PR (if any):_

- Fixes #1518

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by miquelbeltran** [comment]
Looks like I forgot how code excerpts work, will fix and come back

**Review by atsansone — COMMENTED** [review]
@miquelbeltran : There's an error being thrown. Clear that and I can approve.

**Comment by parlough** [comment]
I haven't taken a look at this, but for your future reference, feel free to create your branches directly on `flutter/website`. It allows for automatic staging to make your life easier :)

**Comment by miquelbeltran** [comment]
Sorry for the ci issues! the PR should be ready now

**Review by sfshaza2 — APPROVED** [review]
I tested the DartPad on Safari and this looks good to me. But don't merge it until @parlough can take a look. Thanks, @miquelbeltran!

File: src/cookbook/design/drawer.md
Comment (parlough) [inline src/cookbook/design/drawer.md:139]:
```suggestion
and displays the corresponding text in the center of the `Scaffold`.
```

File: src/cookbook/design/drawer.md
Comment (parlough) [inline src/cookbook/design/drawer.md:143]:
```suggestion
  For more information on implementing navigation,
  check out the [Navigation][] section of the cookbook.
```

**Review by parlough — APPROVED** [review]
Thanks for this expansion. I appreciate the pointing to where to learn more!

**Comment by domesticmouse** [comment]
Looks like there is some deprecated warnings.
```
   info • 'VideoPlayerController.network' is deprecated and shouldn't be used. Use VideoPlayerController.networkUrl instead • lib/main.dart:38:19 • deprecated_member_use
```

I'll leave it to your discretion as to whether to fix this in this PR or a separate PR.

**Pull request merged**
